src/main/java/it/pagopa/pagopa/apiconfig/service/CdiService.java

```diff
@@ -161,6 +172,62 @@ public void deleteCdi(String idCdi, String pspCode) {
         cdiMasterRepository.delete(cdiMaster);
     }
 
+
+    public void uploadHistory() {
+        var result = new ArrayList<CdiCosmos>();
+        cdiMasterValidRepository.findAll()
+                .stream()
+                .filter(Objects::nonNull)
+                .forEach(master -> {
+                    var cdiDetails = master.getCdiDetail() != null ?
+                            master.getCdiDetail()
+                                    .stream()
+                                    .filter(Objects::nonNull)
+                                    .map(this::mapDetails)
+                                    .collect(Collectors.toList())
+                            : null;
+                    result.add(CdiCosmos.builder()
+                            .idPsp(master.getFkPsp().getIdPsp())
+                            .idCDI(master.getIdInformativaPsp())
+                            .cdiStatus("NEW")
+                            .digitalStamp(master.getMarcaBolloDigitale())
+                            .validityDateFrom(master.getDataInizioValidita() != null ? master.getDataInizioValidita().toLocalDateTime() : null)
+                            .details(cdiDetails)
+                            .build());
+                });
+        cdiCosmosRepository.saveAll(result);
+    }
+
+    private CdiDetailCosmos mapDetails(@NotNull CdiDetail detail) {
+        @NotNull Canali canale = detail.getFkPspCanaleTipoVersamento().getCanaleTipoVersamento().getCanale();
+        return CdiDetailCosmos.builder()
+                .idChannel(canale.getIdCanale())
+                .name(detail.getNomeServizio())
+                .description(getDescription(detail))
+                .channelApp(detail.getCanaleApp() == 1L)
+                .paymentMethod(detail.getModelloPagamento())
```

Comment:
are we sure it should be `paymentMethod` and not `paymentType`?
Marketplace knows `paymentType`

Reply:
I can refactor this param